### PR TITLE
Drag from Attribute Editor the graph exception resolution

### PR DESCRIPTION
**Pull request by antares1470**
### Description of the Change
Added a catch for ClassCastException to account for the issue where you would get the exception when you drag a label into the graph.

### Alternate Designs

Looked at options around handling the case when casting ByteBufferInputStream to ByteBuffer (the case that was causing the exception) but ByteBufferInputStream is contained in a private which we can't edit.

Also looked at other options around what to do in the catch block for ClassCastException but ultimately doing nothing felt right.

### Verification Process

- opened a new graph and the attribute editor
- left-clicked and dragged an attribute label from the attribute editor to the graph space
- right-clicked and dragged an attribute label from the attribute editor to the graph space

### Applicable Issues
#448

### CODE REVIEW

**Comment by mimosa2** [comment]
I have tested this. It doesn't throw an exception anymore.

File: CoreAttributeEditorView/src/au/gov/asd/tac/constellation/views/attributeeditor/LabelDropper.java [hunk 1]
```diff
@@ -133,6 +133,8 @@ public void edit(final GraphWriteMethods wg, final PluginInteraction interaction
 
             } catch (final UnsupportedFlavorException | IOException | ClassNotFoundException ex) {
                 Exceptions.printStackTrace(ex);
+            } catch (final ClassCastException ex) {
+                //Do nothing
```
Comment (arcturus2) [inline CoreAttributeEditorView/src/au/gov/asd/tac/constellation/views/attributeeditor/LabelDropper.java:137]: @antares1470 would you be able to explain a bit more of some of the scenarios that lead to this happening "e.g. dropping a attribute label causes this"; something like that so we know why we are doing nothing? Thank you.

Comment (antares1470) [reply CoreAttributeEditorView/src/au/gov/asd/tac/constellation/views/attributeeditor/LabelDropper.java:137]: Done

**Comment by antares1470** [comment]
@cygnus-x-1 `DataFlavor` comes from a Java library so I'm unable to edit anything in it.

As for the code itself, the issue is in line 82:
`transferable.getTransferData(ATTRIBUTE_LABEL_FLAVOR)` has a return type of Object and it is dynamically being decided that it is a `ByteBufferInputStream` which happens to be defined in a private class `DataFlavorUtils` (which isn't our code). This makes it difficult to come up with a better solution than what I went with.

**Comment by cygnus-x-1** [comment]
> @cygnus-x-1 `DataFlavor` comes from a Java library so I'm unable to edit anything in it.
> 
> As for the code itself, the issue is in line 82:
> `transferable.getTransferData(ATTRIBUTE_LABEL_FLAVOR)` has a return type of Object and it is dynamically being decided that it is a `ByteBufferInputStream` which happens to be defined in a private class `DataFlavorUtils` (which isn't our code). This makes it difficult to come up with a better solution than what I went with (as all the better ones I thought of required `ByteBufferInputStream` being publicly accessible).

I might be missing something here - but if that's the case then drag and drop for Attribute Editor labels will always fail and there's no point keeping these classes around. I would have thought there must be a way to update the definition of the `Transferrable` coming from the Attribute Editor to either return the correct object type or change its definition to use `ByteBufferInputStream`?

**Comment by antares1470** [comment]
> I might be missing something here - but if that's the case then drag and drop for Attribute Editor labels will always fail and there's no point keeping these classes around. I would have thought there must be a way to update the definition of the `Transferrable` coming from the Attribute Editor to either return the correct object type or change its definition to use `ByteBufferInputStream`?

The problem with the latter idea is that because it's a private class in a library we don't control, we can't really play around with it in code (someone let me know if that's not true).

If we can't get it to work though, you make a good point about whether we still need LabelDropper. Looking at its usage, there doesn't appear to be other classes using it directly and so it could be removed with little impact on the rest of the code base and may be the better option if can't determine a way to get this working as intended. Interested in thoughts from others. @arcturus2 @algol60

**Comment by arcturus2** [comment]
I'm going to merge this in to fix the Exception. If we want to make some use of this like create nodes then that can still happen later on.

**Pull request merged**